py3status/modules/gazelle.py

```diff
@@ -0,0 +1,153 @@
+            if abs(num) < 1024.0:
+                return "%3.1f %s" % (num, unit)
+            num /= 1024.0
+        return "%.1f%s" % (num, 'YiB')
+
+    def _login(self):
+        self.py3.request(
+                self.baseurl+"/login.php",
+                data={'username': self.username, 'password': self.password},
+                headers={'User-Agent': 'Mozilla/5.0'},
+                cookiejar=self._cookiejar
+                )
+        try:
+            self._index()  # just to try if login info are good
+        except RequestException:
+            self.py3.log(
+                    "Cannot login to %s" % self.baseurl,
+                    level=self.py3.LOG_ERROR
+                    )
+            raise
+        self._cookiejar.save(self.baseurl)
+        self.py3.log(
+                "Saved cookie in storage",
+                level=self.py3.LOG_INFO
+                )
+
+    def _index(self):
+        resp = self.py3.request(
+                self.baseurl+"/ajax.php",
+                params={'action': "index"},
+                headers={
+                    'User-Agent': 'Mozilla/5.0'
+                    },
+                cookiejar=self._cookiejar
+                )
+        return resp.json()
+
+    def post_config_hook(self):
+        self._cookiejar = self.py3.storage_new_cookiejar()
+        try:
+            self._cookiejar.load(self.baseurl)
+            self._index()  # just to try if saved cookie is good
+        except RequestException:
+            self.py3.log(
+                    "Cookie invalid, outdated or missing. Re-logging.",
+                    level=self.py3.LOG_INFO
+                    )
+            self._login()
+
+    def gazelle(self):
+        index = self._index()
+        if index['status'] != "success":
+            msg = "Tracker failed to give me user info"
+            self.py3.log(msg, level=self.py3.LOG_ERROR)
+            raise RequestException(msg)
+        result = index['response']
+        data = {
+            'ratio': result['userstats']['ratio'],
+            'requiredratio': result['userstats']['requiredratio'],
+            'download': self._readable_size(result['userstats']['downloaded']),
+            'upload': self._readable_size(result['userstats']['uploaded']),
+            'messages': result['notifications']['messages'],
+            'notifs': result['notifications']['notifications']
+            }
+
+        # Ok, so if you think that what I'll do next stinks a bit, remember that
+        # the color threshold design in py3status is a gigantic turd.
+        # Main reason of this ugliness is the impossibility to natively define a
+        # variable threshold, so I have to replace keyword by the actual value.
+        for n, item in enumerate(self.thresholds['ratio']):
+            if item[0] == "requiredratio":
+                lst = list(item)
+                lst[0] = data['requiredratio']
+                self.thresholds['ratio'][n] = tuple(lst)
```

Comment:
(Small nit) This is used only on first loop. You said your required ratio could change later meaning `requiredratio` could be changed anytime... or maybe much faster if you suck at seeding. `i3-msg restart` would fix it (gives it new ratio). Just pointing out a small thing.

Reply:
Yes, I need to keep the position of the variable threshold in a variable somewhere.